### ISSUE TITLE
[LOW] [Groups] [$500] Workspace - Addition of Main Email as Workspace Members

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#38043]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.50-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** abebemiherat@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1, Create a new Expensify account
2. Create a workspace and navigate to the workspace chat room from the left-hand navigation (LHN). Click on the header and proceed to the members section

## Expected Result:
The main account should not be added multiple times as a member

## Actual Result:
The main account email is added multiple times as a member

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/49c422a9-ee43-4aa5-a207-07fd848960d4

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019558297473ee277e</li>
        <li>Upwork Job ID: 1767222881317019648</li>
        <li>Last Price Increase: 2024-03-18</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@isabelastisser  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by isabelastisser** [comment]
Hey @zanyrenney, can you please review this and confirm this is related to #wave8? Thanks!

**Comment by jeremy-croff** [comment]
Unable to reproduce here from a fresh account.

**Comment by Santhosh-Sellavel** [comment]
> This behaviour is not only related to creating new workspace but when testing I found some inconsistency in repetition of the main email. Sometime it is repeated 7, 14 or 9(And this number changes when roaming around other channels)

We should find out why this happens & fix it.

**Comment by ikevin127** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue

The main account / workspace admin appears as multiple duplicates of the same user within Workspace -> Members.

### What is the root cause of that problem?

> [!note]
> I was not able to reproduce this on any environment (local dev / staging / production).

This being said -> `visibleChatMemberAccountIDs` should not contain duplicated values in the first place which leads me to think that the root cause of this issue is BE related. Proof to support this claim is that if we look at the report's `participantAccountIDs` we can see duplicates:

<img width="239" alt="Screenshot 2024-03-12 at 18 58 36" src="https://github.com/Expensify/App/assets/56457735/6f76b890-f647-4b27-832d-e09eb4071109">

Keep in mind -> ideally this should not happen from a BE point of view, we should not have FE code to remove duplicates if everything would work as expected from a BE (pusher event) perspective.

In our case what happens is we're using [`ReportUtils.getParticipantsIDs()`](https://github.com/Expensify/App/blob/767bb3c0ffb5e47c559bd062b64678fa8538027c/src/libs/ReportUtils.ts#L4885C10-L4885C28) to get the members count / data which simply returns `visibleChatMemberAccountIDs` as it comes from the BE without filtering out duplicates.

This can lead to edge cases like this issue where if the `report.visibleChatMemberAccountIDs` array can be populated with duplicated IDs by any BE endpoint / pusher event -> the FE simply displaying all the data (including duplicates) within the Workspace -> Members list.

### What changes do you think we should make in order to solve the problem?

We can notice that within the [`ReportUtils.getParticipantsIDs()`](https://github.com/Expensify/App/blob/767bb3c0ffb5e47c559bd062b64678fa8538027c/src/libs/ReportUtils.ts#L4885C10-L4885C28) function, if `isMoneyRequestReport(report)` is true we're removing the duplicates then return `onlyUnique` instead of `visibleChatMemberAccountIDs` (which can contain duplicates).

My proposed FE solution is to use the same principle mentioned above -> check if `isPolicyExpenseChat(report)` (whether the provided report is a Policy Expense chat) -> then return the array without duplicated values making sure to spread `visibleChatMemberAccountIDs` within the Set like so:

```ts
    // ...existing code

    if (isPolicyExpenseChat(report)) {
        const onlyUnique = [...new Set([...visibleChatMemberAccountIDs])];
        return onlyUnique;
    }

    // ...existing code
```
Doing this we ensure that we're not altering the `ReportUtils.getParticipantsIDs()` function beyond what we need for fixing this issue.

**Comment by ikevin127** [comment]
@usman-ghani564 Can you post a video of you reproducing the issue ?

I'm curious of the steps you performed - since trying the actions performed stated in the issue OP did not reproduce it on my side 🤔

**Comment by Santhosh-Sellavel** [comment]
@ikevin127 I'm confused 
You've said
> I was not able to reproduce this on any environment (local dev / staging / production).

And said
> Proof to support this claim is that if we look at the report's participantAccountIDs we can see duplicates:

If you can see duplicates `participantAccountIDs`, how you were not able to reproduce it?

**Comment by ikevin127** [comment]
@Santhosh-Sellavel I found duplicates within `report.participantAccountIDs` which is proof that the duplicates issue can occur on other `report` objects.

For us to reproduce the issue we would need the duplicates to exist within `report.visibleChatMemberAccountIDs`, which I'm pretty sure none of us were able to reproduce yet (unless @usman-ghani564 has video proof w/ Networking tab open) - which leads me to believe that this is BE (pusher event) related, as detailed in my RCA.

**Comment by Santhosh-Sellavel** [comment]
@isabelastisser 

Can you add `internal` label here, as this seems from BE

**Comment by isabelastisser** [comment]
Not sure why an Expensify engineer was not assigned after I applied the Internal label.

**Comment by isabelastisser** [comment]
[asked in Slack.](https://expensify.slack.com/archives/C01SKUP7QR0/p1711045021337569)

**Comment by isabelastisser** [comment]
I couldn't prioritize this yet.

**Comment by isabelastisser** [comment]
Added to VIP.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
@Santhosh-Sellavel can you please provide an update? Thanks!

**Comment by Santhosh-Sellavel** [comment]
> [asked in Slack.](https://expensify.slack.com/archives/C01SKUP7QR0/p1711045021337569)

I was waiting on this, but its marked as internal now waiting for internal engineer to pick this up

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by isabelastisser** [comment]
@Santhosh-Sellavel can you please provide an update? Can you please check if this can still be reproduced? Thanks!

**Comment by isabelastisser** [comment]
@Santhosh-Sellavel, please provide an update! I will DM you for visibility. Thanks!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by Santhosh-Sellavel** [comment]
@isabelastisser Not reproducible

https://github.com/Expensify/App/assets/85645967/50bc5dc7-cf0c-4ba9-aad6-11cfe55c8aae

**Comment by isabelastisser** [comment]
Thanks for confirming, @Santhosh-Sellavel ! Closing this.
